### PR TITLE
add poc to create sig file

### PR DESCRIPTION
**Pull request by mr-tz**
a POC to ease the creation of sig files based on tarballs containing `.pat` files like we have in the `data/` directory here

### CODE REVIEW

File: create_sig.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    )
+    parser.add_argument(
+        "outdir",
+        type=str,
+        help="path to output directory",
+    )
+    parser.add_argument("-e", "--exclude", nargs="*", help="exclude paths that contain exclusion string")
+    parser.add_argument("-i", "--include_pats", nargs="*", help="include pat files that contain inclusion string")
+    parser.add_argument("-d", "--debug", action="store_true", help="enable debugging output on STDERR")
+    parser.add_argument("-s", "--sigfile", default="flare.sig", help="name of output sigfile")
```
Comment (williballenthin) [inline create_sig.py]: lets add `--patfile` so its easy to recover the source data used to generate the `.sig`?

**Comment by williballenthin** [comment]
we can run this in CI to build releases of these signatures (or do it manually and upload to the release page).

File: create_sig.py (hunk 1 above)
Comment (mr-tz) [reply create_sig.py]: Do you mean to explicitly specify `--patfile ./data/VS9/VC/lib/libcmt.lib.pat ./data/VS9/VC/lib/libcmtd.lib.pat ...`?

**Comment by mr-tz** [comment]
```
$ python3 create_sig.py -h
usage: create_sig.py [-h] [-e [EXCLUDE [EXCLUDE ...]]] [-i [INCLUDE_PATS [INCLUDE_PATS ...]]] (--tarballs-root TARBALLS_ROOT | --patfiles PATFILES [PATFILES ...]) [-s SIGFILE] [-d] [-q] outdir

positional arguments:
  outdir                path to output directory

optional arguments:
  -h, --help            show this help message and exit
  -e [EXCLUDE [EXCLUDE ...]], --exclude [EXCLUDE [EXCLUDE ...]]
                        exclude paths that contain exclusion string
  -i [INCLUDE_PATS [INCLUDE_PATS ...]], --include_pats [INCLUDE_PATS [INCLUDE_PATS ...]]
                        include pat files that contain inclusion string
  --tarballs-root TARBALLS_ROOT
                        data path to root of tarballs, use with -e and -i
  --patfiles PATFILES [PATFILES ...]
                        paths to input .pat files
  -s SIGFILE, --sigfile SIGFILE
                        name of output .sig file
  -d, --debug           enable debugging output on STDERR
  -q, --quiet           disable all output but errors

create a .sig file from .pat files stored in tarballs
optionally excludes tarball paths and includes pat file names

example runs:
  $ python3 create_sig.py -d -e libraries test -i libc msvc --tarballs-root data/ -- outdir/
    -d - debug output
    -e - exclude paths containing string `libraries` or `test`
    -i - include pat files containing `libc` and `msvc`
    --tarballs-root - root path storing tarballs containing .pat files
    --
    data/ - root path of tarballs
    outdir/ - path to output files

  $ python3 create_sig.py --patfiles data/VS6/vc98/lib/libc*.pat -- outdir/
    --patfiles - paths to .pat files
    --
    outdir/ - path to output files

```

File: create_sig.py (hunk 1 above)
Comment (mr-tz) [reply create_sig.py]: added in 3eb6ffb, providing two ways now to specify .pat files

**Review by williballenthin — APPROVED** [review]

**Pull request merged**